subsys/bluetooth/mesh/provisionee.c

```diff
@@ -657,6 +659,12 @@ static void prov_link_closed(enum prov_bearer_link_status status)
 	}
 
 	bt_mesh_prov_reset_state();
+
+	/* Re-enable PB-GATT advs if they were stopped due to a PB-ADV link. */
+	if (pb_adv_disabled_pb_gatt && !bt_mesh_is_provisioned()) {
+		bt_mesh_prov_enable(BT_MESH_PROV_GATT);
```

Comment:
besides calling this seems like not safe since it will call [bt_mesh_adv_enable](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/adv_ext.c#L520) through PB-GATT [link_accept](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/pb_gatt.c#L183).

Reply:
Hmm I guess we could use just:
(void)bt_mesh_pb_gatt_srv_enable();
bt_mesh_adv_gatt_update();

(Think we might need the update part to start sending?)

Reply:
The idea was that bt_mesh_pb_gatt_srv_disable was corresponding to bt_mesh_prov_disable, complementary to bt_mesh_prov_enable (but this would return early due to an existing provisioning link)

Reply:
> Hmm I guess we could use just: (void)bt_mesh_pb_gatt_srv_enable(); bt_mesh_adv_gatt_update();
> 
> (Think we might need the update part to start sending?)

Sounds right!

> The idea was that bt_mesh_pb_gatt_srv_disable was corresponding to bt_mesh_prov_disable, complementary to bt_mesh_prov_enable (but this would return early due to an existing provisioning link)

bt_mesh_prov enabled/disable might do the job here but it is not intended to do that, at least this is what I assume. Eventually is it not safe to call here anyway.

Reply:
Sorry for partially delayed comment:

I wonder why bt_mesh_adv_gatt_update() is not called inside bt_mesh_pb_gatt_srv_enable, seems like this is also wrong since it depends the return of bt_mesh_pb_gatt_srv_enable. Maybe we should move bt_mesh_adv_gatt_update to  bt_mesh_pb_gatt_srv_enable as it is done in bt_mesh_pb_gatt_srv_disable.

Reply:
In this case I could just use bt_mesh_pb_gatt_srv_disable and bt_mesh_pb_gatt_srv_enable (and enable already checks whether the device has been provisioned). But I don't think we then need a suspend/resume functionality in pb_gatt as we only want to stop the server?

Reply:
it is you decision, I'd prefer to just represent the actual intention with inline call etc. , which is suspending the PG-GATT until link expires successful or unsuccessful and resume if it is necessary. I am ok with your decision. 👍